test/unit/specs/components/Personalization/utils/surfaceUtils.spec.js

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  buildPageSurface,
+  isPageWideSurface,
+  normalizeSurfaces
+} from "../../../../../../src/components/Personalization/utils/surfaceUtils";
+
+let pageLocation;
+let logger;
+
+const getPageLocation = () => pageLocation;
+
+describe("Personalization::surfaceUtils", () => {
+  beforeEach(() => {
+    logger = jasmine.createSpyObj("logger", ["error", "warn"]);
+    pageLocation = {
+      host: "domain.com",
+      pathname: "/products/test/"
+    };
+  });
+
+  it("builds page-wide surface from location", () => {
+    expect(buildPageSurface(getPageLocation)).toEqual(
+      "web://domain.com/products/test"
+    );
+    pageLocation = {
+      host: "DOMain.com",
+      pathname: undefined
+    };
+    expect(buildPageSurface(getPageLocation)).toEqual("web://domain.com/");
+    pageLocation = {
+      host: "domain.com:8080",
+      pathname: "/"
+    };
+    expect(buildPageSurface(getPageLocation)).toEqual("web://domain.com:8080/");
+    pageLocation = {
+      host: "domain.com",
+      pathname: "/a"
+    };
+    expect(buildPageSurface(getPageLocation)).toEqual("web://domain.com/a");
+    pageLocation = {
+      host: "domain.com",
+      pathname: "/a/"
+    };
+    expect(buildPageSurface(getPageLocation)).toEqual("web://domain.com/a");
+  });
+
+  it("checks for a page-wide surface", () => {
+    expect(isPageWideSurface("__view__")).toBeFalse();
+    expect(isPageWideSurface("name")).toBeFalse();
+    expect(isPageWideSurface("web://domain.com")).toBeTrue();
+    expect(isPageWideSurface("web://domain.com/path/page.html")).toBeTrue();
+    expect(isPageWideSurface("web://domain.com#fragment")).toBeFalse();
+    expect(isPageWideSurface("webapp://domain.com")).toBeFalse();
+    expect(isPageWideSurface("webapp://domain.com#view")).toBeFalse();
+  });
+
+  it("expands fragment surfaces", () => {
```

Comment:
As mentioned earlier `normalizeSurfaces` could be much simpler if we do not use dep injection.

Reply:
Replied in the earlier comment